Add support for transactions

WhoSoup opened the pull request:
In a similar vein to #140, this changes the factom api to return a struct containing the data rather than just returning a JSON string. Includes a basic unit test.

WhoSoup commented:
Decided to add support for decoding the transactions in the FBlock as well, rather than just having them be raw json.

PaulBernier commented on transaction_pending.go:
```diff
@@ -0,0 +1,40 @@
+		return nil, err
+	}
+	if resp.Error != nil {
+		return nil, err
```
return resp.Error? Because err is nil at this point

WhoSoup replied:
Good catch

PaulBernier commented on fblock.go:
```diff
@@ -8,24 +8,152 @@ import (
+	Signatures []string `json:"signatures"`
+}
+
+func (t *FBTransaction) MarshalJSON() ([]byte, error) {
```
are MarshalJSON and UnmarshalJSON used anywhere?
And can you add unit test for those methods?

WhoSoup replied:
They are used recursively by Marshalling/Unmarshalling FBlock.Transactions. UnmarshalJSON has unit test coverage due to TestUnmarshalFBlock but it appears MarshalJSON does not, which I'll add.

WhoSoup commented on fblock.go:
```diff
@@ -8,24 +8,152 @@ import (
+		return fmt.Errorf("invalid signature counts")
+	}
+
+	for i := range txResp.Inputs {
```
I needed the index anyway to access the respective `tx.RCDs` and `tx.SigBlocks`, so I decided to just use the index for all three to make the association clearer.

WhoSoup commented:
Addressed the comments. Great call on the MarshalJSON unit test, managed to catch a bug where it didn't convert the timestamp properly back to millitimestamp.
